src/Features/LanguageServer/Protocol/Handler/Diagnostics/DocumentPullDiagnosticHandler.cs

```diff
@@ -119,7 +119,7 @@ protected override DiagnosticTag[] ConvertTags(DiagnosticData diagnosticData)
 
             if (!context.IsTracking(document.GetURI()))
             {
-                context.TraceWarning($"Ignoring diagnostics request for untracked document: {document.GetURI()}");
+                context.TraceWarning($"Ignoring diagnostics request for untracked document");
```

Comment:
How does this help? If GetURI would have thrown, wouldn't it already have crashed on line 120?

Reply:
also, can we move away from GetURI ever crashing?  It's unclear to me why it ever would, and it seems bad as that indicates that there are roslyn documents that we cannot discuss with LSP.

Reply:
Do we have any other properties could be used here to give hints about the warning?

Reply:
> also, can we move away from GetURI ever crashing? It's unclear to me why it ever would, and it seems bad as that indicates that there are roslyn documents that we cannot discuss with LSP.

This! The only thing off the top of my head would be something we can't make a URI for would be source generated files, but we're already tracking that as a bug and if that was the problem here we shouldn't be making changes. But the bug @akhera99 has shared to me privately looks like it's something entirely different.